integrationTests/relayers/slowTests/framework/multiversxHandler.go

```diff
@@ -1113,6 +1143,104 @@ func (handler *MultiversxHandler) scCallAndCheckTx(
+	hash := ""
+	handler.TestHelperAddress, hash, _ = handler.ChainSimulator.DeploySC(
+		ctx,
+		testHelperContractPath,
+		handler.OwnerKeys.MvxSk,
+		deployGasLimit,
+		make([]string, 0),
+	)
+	require.NotEqual(handler, emptyAddress, handler.TestHelperAddress)
+	log.Info("Deploy: Test helper contract", "address", handler.TestHelperAddress, "transaction hash", hash)
+}
+
+// SetBridgeProxyAddressOnHelper will set the bridge proxy address on the helper contract
+func (handler *MultiversxHandler) SetBridgeProxyAddressOnHelper(ctx context.Context) {
+	params := []string{
+		hex.EncodeToString(handler.ScProxyAddress.Bytes()),
+	}
+	hash, txResult := handler.scCallAndCheckTx(
+		ctx,
+		handler.OwnerKeys,
+		handler.TestHelperAddress,
+		zeroStringValue,
+		setCallsGasLimit,
+		setCalleeFunction,
+		params)
+
+	log.Info("set bridge proxy address to helper", "hash", hash, "status", txResult.Status)
+}
+
+// CallDepositOnBridgeProxy will call the callDeposit function from test helper contract which will call the deposit function on the bridge proxy
+func (handler *MultiversxHandler) CallDepositOnBridgeProxy(ctx context.Context, ethTx EthTransaction, batchId uint64, token *TokenData) {
+	buffNonce := make([]byte, 8)
+	binary.BigEndian.PutUint64(buffNonce, ethTx.Nonce)
+
+	buffBatchId := make([]byte, 8)
+	binary.BigEndian.PutUint64(buffBatchId, batchId)
+
+	params := []string{
+		hex.EncodeToString([]byte(token.MvxUniversalToken)),
+		hex.EncodeToString(ethTx.Amount.Bytes()),
+		hex.EncodeToString([]byte(callDepositFunction)),
+		hex.EncodeToString(buffBatchId),
+		hex.EncodeToString(ethTx.From),
+		hex.EncodeToString(ethTx.To),
+		hex.EncodeToString([]byte(ethTx.TokenID)),
+		hex.EncodeToString(ethTx.Amount.Bytes()),
+		hex.EncodeToString(buffNonce),
+		hex.EncodeToString(ethTx.CallData),
+	}
+
+	hash, txResult := handler.scCallAndCheckTx(
+		ctx,
+		handler.OwnerKeys,
+		handler.TestHelperAddress,
+		zeroStringValue,
+		generalSCCallGasLimit,
+		esdtTransferFunction,
+		params)
+
+	log.Info("deposit on bridge proxy tx executed", "hash", hash, "status", txResult.Status)
+}
+
+// ExecuteDepositWithoutGenerateBlocks will execute the deposit on the bridge proxy without generating blocks
+func (handler *MultiversxHandler) ExecuteDepositWithoutGenerateBlocks(ctx context.Context, depositId uint64, txNonce uint64) {
+	buffDepositId := make([]byte, 8)
+	binary.BigEndian.PutUint64(buffDepositId, depositId)
+
+	params := []string{
+		executeFunction,
+		hex.EncodeToString(buffDepositId),
+	}
+	txData := strings.Join(params, "@")
+
+	hash := handler.ChainSimulator.SendTxWithoutGenerateBlocksAndNonce(
+		ctx,
+		handler.SCExecutorKeys.MvxSk,
+		handler.ScProxyAddress,
+		txNonce,
+		zeroStringValue,
+		generalSCCallGasLimit,
+		[]byte(txData))
+
+	log.Info("execute deposit on bridge proxy tx sent", "transaction hash", hash)
+
+	txResult, txStatus := handler.ChainSimulator.GetTransactionResult(ctx, hash)
```

Comment:
out of context code (L1232-L1242) should not be present here because:
1. the name of the function does not reflect that these checks are carried
2. reduces the reusability of the code: if we will going to reuse this function, it will complicate the checks
3. calling `GetTransactionResult` here defeats the purpose of the test as it generates blocks until the transaction is executed